tests/paramgrill.c

```diff
@@ -240,45 +368,63 @@ static void findClockGranularity(void) {
     DEBUGOUTPUT("Granularity: %llu\n", (unsigned long long)g_clockGranularity);
 }
 
+/* allows zeros */
 #define CLAMPCHECK(val,min,max) {                     \
-    if (val && (((val)<(min)) | ((val)>(max)))) {     \
+    if (((val)<(min)) | ((val)>(max))) {              \
         DISPLAY("INVALID PARAMETER CONSTRAINTS\n");   \
         return 0;                                     \
 }   }
 
 
 /* Like ZSTD_checkCParams() but allows 0's */
 /* no check on targetLen? */
-static int cParamValid(ZSTD_compressionParameters paramTarget) {
-    CLAMPCHECK(paramTarget.hashLog, ZSTD_HASHLOG_MIN, ZSTD_HASHLOG_MAX);
-    CLAMPCHECK(paramTarget.searchLog, ZSTD_SEARCHLOG_MIN, ZSTD_SEARCHLOG_MAX);
-    CLAMPCHECK(paramTarget.searchLength, ZSTD_SEARCHLENGTH_MIN, ZSTD_SEARCHLENGTH_MAX);
-    CLAMPCHECK(paramTarget.windowLog, ZSTD_WINDOWLOG_MIN, ZSTD_WINDOWLOG_MAX);
-    CLAMPCHECK(paramTarget.chainLog, ZSTD_CHAINLOG_MIN, ZSTD_CHAINLOG_MAX);
-    if(paramTarget.targetLength > ZSTD_TARGETLENGTH_MAX) {
-        DISPLAY("INVALID PARAMETER CONSTRAINTS\n");
-        return 0;
-    }
-    if(paramTarget.strategy > ZSTD_btultra) {
-        DISPLAY("INVALID PARAMETER CONSTRAINTS\n");
-        return 0;
+static int paramValid(paramValues_t paramTarget) {
+    U32 i;
+    for(i = 0; i < NUM_PARAMS; i++) {
+        CLAMPCHECK(paramTarget.vals[i], mintable[i], maxtable[i]);
     }
+    //TODO: Strategy could be valid at 0 before, is that right?
```

Comment:
answering the comment : 
yes, it used to be true, `ZSTD_fast` used to be `==0`,
but it changed some time ago, and now `0` is no longer a valid value for this `enum`.